src/main/java/org/jfrog/filespecs/aql/AqlBuildingUtils.java

```diff
@@ -86,13 +86,20 @@ private static String prepareSearchPattern(String pattern) {
         return pattern.replaceAll("[()]", "");
     }
 
-    private static String buildExcludeQuery(String[] exclusions, boolean useLocalPath, boolean recursive) {
-        if (ArrayUtils.isEmpty(exclusions)) {
+    private static String buildExcludeQuery(String[] exclusions, String[] excludePatterns, boolean useLocalPath, boolean recursive) {
+        if (ArrayUtils.isEmpty(exclusions) && ArrayUtils.isEmpty(excludePatterns)) {
             return "";
         }
         List<RepoPathFile> excludeTriples = new ArrayList<>();
-        for (String exclusion : exclusions) {
-            excludeTriples.addAll(PatternParsingUtils.createRepoPathFileTriples(prepareSearchPattern(exclusion), recursive));
+        if (exclusions != null && exclusions.length > 0) {
```

Comment:
This line can be shortened to -
```if (exclusions != null)```
The length part is already being taken care fo by the following loop.